tests/ConnectionTest.php

```diff
@@ -347,4 +349,21 @@ public function testEventListenOrder(): void
         $this->assertEquals(QueryExecuted::class, $receivedEventClasses[1]);
         $this->assertEquals(TransactionCommitted::class, $receivedEventClasses[2]);
     }
+
+    public function test_Connection_transaction_reset_on_exceptions(): void
+    {
+        $conn = $this->getDefaultConnection();
+
+        try {
+            $conn->transaction(function () {
```

Comment:
```suggestion
            $conn->transaction(function (Connection $conn) {
                self::assertTrue($conn->inTransaction());
                self::assertNotNull($conn->getCurrentTransaction());
                self::assertSame(1, $conn->transactionLevel());
```
How about asserting before throwing an exception to make sure that the transaction is definitely initiated?